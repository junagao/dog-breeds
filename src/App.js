import React from 'react';
import { hot } from 'react-hot-loader/root';

const App = () => <div>hello world</div>;

export default hot(App);
