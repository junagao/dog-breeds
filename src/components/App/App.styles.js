import styled from 'styled-components';

const AppContainer = styled.div`
  padding: 2rem;
  width: 90%;
  margin: 0 auto;
  text-align: center;
`;

export default AppContainer;
