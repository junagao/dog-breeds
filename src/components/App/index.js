export { connectedApp as default } from './App';
