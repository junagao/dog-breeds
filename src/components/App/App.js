import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader/root';
import { connect } from 'react-redux';
import { getBreeds, getImages } from 'actions/dogs';
import { Chart, Loading } from 'components';
import AppContainer from './App.styles';

export class App extends React.Component {
  componentDidMount() {
    const { getBreeds } = this.props;
    getBreeds();
  }

  componentDidUpdate(prevProps) {
    const { getImages, breeds } = this.props;
    if (breeds !== prevProps.breeds) {
      getImages();
    }
  }

  render() {
    const { dogs, loading } = this.props;

    return (
      <div>
        {loading ? (
          <Loading />
        ) : (
          <AppContainer>
            <Chart dogs={dogs} />
          </AppContainer>
        )}
      </div>
    );
  }
}

App.propTypes = {
  getBreeds: PropTypes.func.isRequired,
  getImages: PropTypes.func.isRequired,
  breeds: PropTypes.arrayOf(PropTypes.string).isRequired,
  dogs: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  loading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  breeds: state.breeds,
  dogs: state.dogs,
  loading: state.loading,
  error: state.error,
});

const mapDispatchToProps = {
  getBreeds,
  getImages,
};

const connectedApp = hot(connect(mapStateToProps, mapDispatchToProps)(App));

export default connectedApp;
