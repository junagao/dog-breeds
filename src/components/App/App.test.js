import React from 'react';
import { render } from '@testing-library/react';
import { App } from './App';

describe('App', () => {
  const mockBreeds = ['chihuahua', 'collie', 'husky'];
  const mockDogs = [
    {
      breed: 'chihuahua',
      images: [
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10074.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10131.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10621.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_1073.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10976.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_11140.jpg',
      ],
      dogCount: 6,
    },
    {
      breed: 'collie',
      images: [
        'https://images.dog.ceo/breeds/collie-border/n02106166_1031.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1032.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1055.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1056.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1059.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1128.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1133.jpg',
      ],
      dogCount: 7,
    },
    {
      breed: 'husky',
      images: [
        'https://images.dog.ceo/breeds/husky/20180901_150234.jpg',
        'https://images.dog.ceo/breeds/husky/20180904_185604.jpg',
        'https://images.dog.ceo/breeds/husky/20180924_193829.jpg',
      ],
      dogCount: 3,
    },
    {
      breed: 'chihuahua',
      images: [
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10074.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10131.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10621.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_1073.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_10976.jpg',
        'https://images.dog.ceo/breeds/chihuahua/n02085620_11140.jpg',
      ],
      dogCount: 6,
    },
    {
      breed: 'collie',
      images: [
        'https://images.dog.ceo/breeds/collie-border/n02106166_1031.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1032.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1055.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1056.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1059.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1128.jpg',
        'https://images.dog.ceo/breeds/collie-border/n02106166_1133.jpg',
      ],
      dogCount: 7,
    },
    {
      breed: 'husky',
      images: [
        'https://images.dog.ceo/breeds/husky/20180901_150234.jpg',
        'https://images.dog.ceo/breeds/husky/20180904_185604.jpg',
        'https://images.dog.ceo/breeds/husky/20180924_193829.jpg',
      ],
      dogCount: 3,
    },
  ];
  test('matches snapshot', () => {
    const { container } = render(
      <App
        getBreeds={jest.fn()}
        getImages={jest.fn()}
        breeds={mockBreeds}
        dogs={mockDogs}
        loading={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('render an instance of Loading component if loading is true', () => {
    const { getByTestId } = render(
      <App
        getBreeds={jest.fn()}
        getImages={jest.fn()}
        breeds={mockBreeds}
        dogs={mockDogs}
        loading
      />,
    );
    expect(getByTestId('spinner')).toBeInTheDocument();
  });

  test('render an instance of Chart if loading is false', () => {
    const { getByTestId } = render(
      <App
        getBreeds={jest.fn()}
        getImages={jest.fn()}
        breeds={mockBreeds}
        dogs={mockDogs}
        loading={false}
      />,
    );
    expect(getByTestId('pie-chart')).toBeInTheDocument();
  });
});
