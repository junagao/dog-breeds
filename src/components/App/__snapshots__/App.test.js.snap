// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
.c1 {
  color: #444;
  margin-bottom: 2rem;
}

.c2 {
  color: #444;
  margin-top: 2rem;
  font-size: 12px;
}

.c3 {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.25rem;
}

.c3:hover {
  color: #fff;
  background-color: #000;
  border-radius: 0.25rem;
}

.c0 {
  padding: 2rem;
  width: 90%;
  margin: 0 auto;
  text-align: center;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        currentitem="false"
        data-for="chart"
        data-testid="pie-chart"
        data-tip=""
      >
        <h1
          class="c1"
        >
          Images by Dog Breed (%)
        </h1>
        <svg
          height="100%"
          style="height: 600px;"
          viewBox="0 0 100 100"
          width="100%"
        >
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(0,100%,50%)"
            stroke-width="50"
          />
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(3.7978723404255317,100%,50%)"
            stroke-width="50"
          />
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(7.595744680851063,100%,50%)"
            stroke-width="50"
          />
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(11.393617021276595,100%,50%)"
            stroke-width="50"
          />
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(15.191489361702127,100%,50%)"
            stroke-width="50"
          />
          <path
            d="M NaN NaN A 25 25 0 1 1 NaN NaN"
            fill="none"
            stroke="hsl(18.98936170212766,100%,50%)"
            stroke-width="50"
          />
        </svg>
        <div
          class="__react_component_tooltip t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6 place-top type-dark"
          data-id="tooltip"
          id="chart"
        >
          <style>
            
  	.t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-top {
        margin-top: -10px;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-top::before {
        border-top: 8px solid transparent;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-bottom {
        margin-top: 10px;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-left {
        margin-left: -10px;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-left::before {
        border-left: 8px solid transparent;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-right {
        margin-left: 10px;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-right::before {
        border-right: 8px solid transparent;
    }
    .t7c9ed0ad-53cb-449e-ae74-405b00a3fbd6.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
          </style>
        </div>
        <p
          class="c2"
        >
          Data source:
           
          <a
            class="c3"
            href="https://dog.ceo/dog-api/documentation/"
            target="_blank"
          >
            Dog API
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
